### PR TITLE
windows compiling with native tools and clang, macos catalina compiling, added travis ci build tests

### PR DESCRIPTION
**Pull request by gelotus**
Compiles fine in windows with cmake, with clang, clang-cl and cl. Compiles fine in linux with autotools, cmake, clang and gcc. Compiles fine in macos catalina with cmake, autotools and clang. Added travis ci build test of these. I think windows mingw is broken, at least with 8.1 that travis ships. There is no 64bit usblib and i was unable to  tell wingw to take 32bit libs when compiles, but i've not lost much time on this frankly, since msvc build just works. Maybe the next move will be to drop autotools and use only cmake, and migrate to libusb1 with libusb-compat wrapper. That's all for now.

### CODE REVIEW

File: cmake/modules/LibnfcDrivers.cmake [hunk 1]
```diff
@@ -3,13 +3,13 @@ SET(LIBNFC_DRIVER_ACR122_PCSC OFF CACHE BOOL "Enable ACR122 support (Depends on
 SET(LIBNFC_DRIVER_ACR122_USB ON CACHE BOOL "Enable ACR122 support (Direct USB connection)")
 SET(LIBNFC_DRIVER_ACR122S ON CACHE BOOL "Enable ACR122S support (Use serial port)")
 SET(LIBNFC_DRIVER_ARYGON ON CACHE BOOL "Enable ARYGON support (Use serial port)")
-IF(WIN32)
+IF(NOT UNIX OR APPLE)
```
Comment (doegox) [inline cmake/modules/LibnfcDrivers.cmake:6]: May I suggest to reverse the test and branches, to ease reading ?
if (unix or apple) ... set ON... else... set OFF

File: libnfc/CMakeLists.txt [hunk 2]
```diff
@@ -26,23 +29,23 @@ IF(UART_REQUIRED)
 ENDIF(UART_REQUIRED)
 
 IF(I2C_REQUIRED)
-  IF(WIN32)
-    # Windows is not supported at the moment
+  IF(NOT UNIX OR APPLE)
```
Comment (doegox) [inline libnfc/CMakeLists.txt:32]: May I suggest to reverse the test and branches, to ease reading ?
if (unix or apple) ... set ON... else... set OFF

File: libnfc/CMakeLists.txt [hunk 3]
```diff
@@ -26,23 +29,23 @@ IF(UART_REQUIRED)
...
+  ELSE(NOT UNIX OR APPLE)
     LIST(APPEND BUSES_SOURCES buses/i2c)
-  ENDIF(WIN32)
+  ENDIF(NOT UNIX OR APPLE)
 ENDIF(I2C_REQUIRED)
 
 IF(SPI_REQUIRED)
-  IF(WIN32)
-    # Windows is not supported at the moment
+  IF(NOT UNIX OR APPLE)
```
Comment (doegox) [inline libnfc/CMakeLists.txt:42]: May I suggest to reverse the test and branches, to ease reading ?
if (unix or apple) ... set ON... else... set OFF

File: cmake/modules/LibnfcDrivers.cmake (hunk 1 above)
Comment (gelotus) [reply cmake/modules/LibnfcDrivers.cmake:6]: I know it's ugly but there is no LINUX macro in cmake, and APPLE falls into UNIX like LINUX, i could do IF(UNIX AND NOT APPLE). Let me know

Comment (doegox) [reply cmake/modules/LibnfcDrivers.cmake:6]: damn, I managed to read it wrong myself :laughing: 
yes I tend to prefer UNIX AND NOT APPLE

**Comment by doegox** [comment]
very impressive, including Travis, thanks a lot !

**Comment by gelotus** [comment]
You are welcome. I think you can close pr #418 #457 and #478

**Pull request merged**